x-pack/plugins/observability_solution/observability_ai_assistant/server/service/client/operators/with_langtrace_chat_complete_span.ts

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { Event, LLMSpanAttributes } from '@langtrase/trace-attributes';
+import { Span } from '@opentelemetry/api';
+import { FunctionDefinition } from 'openai/resources';
+import { ignoreElements, last, merge, OperatorFunction, share, tap } from 'rxjs';
+import { Message, StreamingChatResponseEventType } from '../../../../common';
+import { ChatEvent } from '../../../../common/conversation_complete';
+import { concatenateChatCompletionChunks } from '../../../../common/utils/concatenate_chat_completion_chunks';
+import { withoutTokenCountEvents } from '../../../../common/utils/without_token_count_events';
+import { getLangtraceSpanAttributes } from '../instrumentation/get_langtrace_span_attributes';
+
+export enum LangtraceServiceProvider {
+  OpenAI = 'OpenAI',
+  Azure = 'Azure',
+  Anthropic = 'Anthropic',
+}
+
+export function withLangtraceChatCompleteSpan<T extends ChatEvent>({
+  span,
+  model,
+  messages,
+  serviceProvider,
+  functions,
+}: {
+  span: Span;
+  model: string;
+  messages: Message[];
+  serviceProvider: LangtraceServiceProvider;
+  functions?: Array<Pick<FunctionDefinition, 'name' | 'description' | 'parameters'>>;
+}): OperatorFunction<T, T> {
+  const attributes: LLMSpanAttributes = {
+    ...getLangtraceSpanAttributes(),
+    'langtrace.service.name': serviceProvider,
+    'llm.api': '/chat/completions',
+    'http.max.retries': 0,
+    'url.full': 'http://localhost:3000/chat/completions',
```

Comment:
localhost? Perhaps a note that this is because it's only meant for dev-use and should be changed in the future if we want to use this in prod.

Reply:
No good catch, I just forgot to fix it